src/Tests/Microsoft.Diagnostics.Monitoring.TestCommon/DotNetHost.cs

```diff
@@ -33,7 +33,15 @@ public static string GetPath(Architecture? arch = null)
                 // e.g. Append "\x86" to the path
                 dotnetDirPath = Path.Combine(dotnetDirPath, arch.Value.ToString("G").ToLowerInvariant());
             }
-            return Path.GetFullPath(Path.Combine(dotnetDirPath, ExeName));
+
+            string absoluteExecutablePath = Path.GetFullPath(Path.Combine(dotnetDirPath, ExeName));
+            if (!File.Exists(absoluteExecutablePath) && RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("VSAPPIDDIR")))
```

Comment:
Can you add comments as to why this fallback is used and why you chose these specific conditions?